django/db/models/expressions.py

```diff
@@ -412,6 +432,12 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def refs_aggregate(self, existing_aggregates):
         return refs_aggregate(self.name.split(LOOKUP_SEP), existing_aggregates)
 
+    def asc(self):
```

Comment:
Nope, because `F()` doesn't inherit from `BaseExpression` and `asc()` is called before `resolve_expression()`.